EventPlanner/app/src/main/java/m3/eventplanner/viewmodel/HomeScreenViewModel.java

```diff
@@ -0,0 +1,274 @@
+                    if (response.body().isEmpty()) {
+                        topEvents.setValue(null);
+                    } else {
+                        topEvents.setValue(new ArrayList<>(response.body()));
+                    }
+                } else {
+                    topEvents.setValue(null);
+                }
+            }
+
+            @Override
+            public void onFailure(Call<Collection<GetEventDTO>> call, Throwable t) {
+                topEvents.setValue(null);
+            }
+        });
+    }
+
+    public void loadTopOfferings() {
+        clientUtils.getOfferingService().getTopOfferings().enqueue(new Callback<Collection<GetOfferingDTO>>() {
+            @Override
+            public void onResponse(Call<Collection<GetOfferingDTO>> call, Response<Collection<GetOfferingDTO>> response) {
+                if (response.isSuccessful() && response.body() != null) {
+                    topOfferings.setValue(new ArrayList<>(response.body()));
+                }
+            }
+
+            @Override
+            public void onFailure(Call<Collection<GetOfferingDTO>> call, Throwable t) {
+                topOfferings.setValue(new ArrayList<>());
+            }
+        });
+    }
+    public void loadPagedEvents(int page) {
+        fetchPagedEvents(page);
+    }
+    public void loadFilteredEvents(int page, Integer eventTypeId, String location, Integer maxParticipants, Double minRating, String startDate, String endDate) {
+        this.filterEventTypeId = eventTypeId;
+        this.filterEventLocation = location;
+        this.filterEventMaxParticipants = maxParticipants;
+        this.filterEventMinRating = minRating;
+        this.filterEventStartDate = startDate;
+        this.filterEventEndDate = endDate;
+
+        fetchPagedEvents(page);
+    }
+    public void loadSearchedEvents(int page, String query){
+        this.eventSearchQuery = query;
+
+        fetchPagedEvents(page);
+    }
+    public void loadSortedEvents(int page, String sortBy, String sortDirection){
+        this.sortEventsBy = sortBy;
+        this.sortEventsDirection = sortDirection;
+
+        fetchPagedEvents(page);
+    }
+
+    private void fetchPagedEvents(int page){
+        currentEventPage = page;
+        currentPaginationContext = PaginationContext.EVENTS;
+
+        clientUtils.getEventService().getEvents(currentEventPage, eventPageSize,
+                filterEventTypeId, filterEventLocation, filterEventMaxParticipants, filterEventMinRating, filterEventStartDate, filterEventEndDate,
+                eventSearchQuery, sortEventsBy, sortEventsDirection).enqueue(new Callback<PagedResponse<GetEventDTO>>() {
+            @Override
+            public void onResponse(Call<PagedResponse<GetEventDTO>> call, Response<PagedResponse<GetEventDTO>> response) {
+                if (response.isSuccessful() && response.body() != null && response.body().getContent() != null) {
```

Comment:
Just a few suggestions:

Instead of calling response.body().getContent() multiple times, we could store it in a variable (eventList) to avoid redundancy and improve readability.
The check for eventList.isEmpty() could be moved inside a single if block for clarity.
We could simplify the flow by using pagedEvents.setValue(null) in both the null and empty list cases.
Totally up to you if you'd like to implement these changes

Reply:
Refactored Code:

    if (response.isSuccessful() && response.body() != null) {
        PagedResponse<GetEventDTO> pagedResponse = response.body();
        List<GetEventDTO> eventList = pagedResponse.getContent();

        if (eventList != null && !eventList.isEmpty()) {
            pagedEvents.setValue(pagedResponse);
            totalEventPages = pagedResponse.getTotalPages();
            totalEvents = pagedResponse.getTotalElements();
        } else {
            pagedEvents.setValue(null);
        }
    } else {
        pagedEvents.setValue(null);
    }